core/src/oracles.ts

```diff
@@ -0,0 +1,145 @@
+    CurrentPriceOnly: async (
+        oracle: OracleCurrentPriceOnly,
+        provider: ethers.providers.JsonRpcProvider,
+        oracleAddress: string
+    ) => {
+        /**
+         * Get current price from the contract via direct memory access (slot address)
+         * The price is stored in separate slot as a validity marker:
+         *   - the marker might be stored in the same slot as some other irrelevent value
+         *   - extract the byte that contains validity marker only
+         * If the price is valid - return it, otherwise provide a NaN
+         **/
+        const currentPriceFeed = await provider.getStorageAt(oracleAddress, oracle.currentPriceSlotAddress);
+        const slotPriceValueBeginsAtPosition = oracle.slotPriceValueBeginsAtPosition;
+        const priceValiditySlotValue = await provider.getStorageAt(
+            oracleAddress,
+            oracle.currentPriceValiditySlotAndOffset.slot
+        );
+        const isPriceValid =
+            parseInt(priceValiditySlotValue[oracle.currentPriceValiditySlotAndOffset.offset], 16) === 1;
+        return isPriceValid
+            ? new BigNumber(`0x${currentPriceFeed.substring(slotPriceValueBeginsAtPosition)}`)
+            : new BigNumber(NaN);
+    },
+    CurrentAndNextPrice: async (
+        oracle: OracleCurrentAndNextPrices,
+        provider: ethers.providers.JsonRpcProvider,
+        oracleAddress: string
+    ) => {
+        return await getOraclePriceSameSlotValidity(
+            oracle.currentPriceSlotAddress,
+            oracle.slotPriceValueBeginsAtPosition,
+            provider,
+            oracleAddress
+        );
+    },
+};
+const getCurrentOraclePrice = async (
+    oracle: CollateralPriceSourceConfig,
+    provider: ethers.providers.JsonRpcProvider,
+    oracleAddress: string
+) => {
+    return currentPriceExtractors[oracle.type](oracle, provider, oracleAddress);
+};
+
+const getNextOraclePriceChange = async (
+    collateralConfig: CollateralConfig,
+    provider: ethers.providers.JsonRpcProvider,
+    oracleAddress: string
+) => {
+    let nextPriceChange: Date = new Date(NaN);
+    if (collateralConfig.oracle.hasDelay) {
+        const osmContractInterface = await getContractInterfaceByName('OSM');
+        const osmContract = new ethers.Contract(oracleAddress, osmContractInterface, provider);
+        const lastPriceUpdateAsHex = (await osmContract.zzz())._hex;
+        const priceUpdateFrequencyInSeconds = await osmContract.hop();
+        const lastPriceUpdateTimestampInSeconds = new BigNumber(lastPriceUpdateAsHex).toNumber();
+        nextPriceChange = new Date((lastPriceUpdateTimestampInSeconds + priceUpdateFrequencyInSeconds) * 1000);
+    }
+    return nextPriceChange;
+};
+
+const _getOsmPrices = async (
+    network: string,
+    oracleAddress: string,
+    collateralType: CollateralType
+): Promise<OraclePrices> => {
+    const provider = await getProvider(network);
+    const collateralConfig = COLLATERALS[collateralType];
```

Comment:
Please prefer using `getCollateralConfigByType` which automatically throws in case there is no such collateral